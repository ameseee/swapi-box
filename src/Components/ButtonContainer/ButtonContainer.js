import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';
import '../../index.css';

const ButtonContainer = () => {

  return (
    <div className='button-container'>
      <Button />
      <Button />
      <Button />
    </div>
  );

};

// ButtonContainer.propTypes = {
//
// };

export default ButtonContainer;
