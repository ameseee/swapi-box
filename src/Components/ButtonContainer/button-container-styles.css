.button-container {
  background: black;
  padding-bottom: 42px; }
