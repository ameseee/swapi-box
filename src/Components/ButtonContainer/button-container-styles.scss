.button-container {
  border: 2px solid red;
}
