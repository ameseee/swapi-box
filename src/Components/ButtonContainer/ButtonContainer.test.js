import React from 'react';
import ButtonContainer from './ButtonContainer';
import { shallow } from 'enzyme';

describe('<ButtonContainer />', () => {
  let wrapper = shallow(<ButtonContainer />);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render a div', () => {
    const div = wrapper.find('.button-container');

    expect(div.length).toEqual(1);
  });

  it('should match the Card snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});
