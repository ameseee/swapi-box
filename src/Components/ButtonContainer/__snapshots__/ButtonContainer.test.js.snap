// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonContainer /> should match the Card snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          handleClick={undefined}
          value="vehicles"
/>,
        <Button
          handleClick={undefined}
          value="people"
/>,
        <Button
          handleClick={undefined}
          value="planets"
/>,
        <Button
          handleClick={undefined}
          value="faves"
/>,
      ],
      "className": "button-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "button-vehicles",
        "nodeType": "function",
        "props": Object {
          "handleClick": undefined,
          "value": "vehicles",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "button-people",
        "nodeType": "function",
        "props": Object {
          "handleClick": undefined,
          "value": "people",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "button-planets",
        "nodeType": "function",
        "props": Object {
          "handleClick": undefined,
          "value": "planets",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "button-faves",
        "nodeType": "function",
        "props": Object {
          "handleClick": undefined,
          "value": "faves",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            handleClick={undefined}
            value="vehicles"
/>,
          <Button
            handleClick={undefined}
            value="people"
/>,
          <Button
            handleClick={undefined}
            value="planets"
/>,
          <Button
            handleClick={undefined}
            value="faves"
/>,
        ],
        "className": "button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "button-vehicles",
          "nodeType": "function",
          "props": Object {
            "handleClick": undefined,
            "value": "vehicles",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "button-people",
          "nodeType": "function",
          "props": Object {
            "handleClick": undefined,
            "value": "people",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "button-planets",
          "nodeType": "function",
          "props": Object {
            "handleClick": undefined,
            "value": "planets",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "button-faves",
          "nodeType": "function",
          "props": Object {
            "handleClick": undefined,
            "value": "faves",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
